bionic/core/flow_execution.py

```diff
@@ -327,7 +340,15 @@ def _wait_on_in_progress_entries(self):
         futures = set(
             entry.future for entry in self._in_progress_entries_by_task_key.values()
         )
-        finished_futures, _ = wait(futures, return_when=FIRST_COMPLETED)
+        if self._parallel_execution_enabled:
+            finished_futures, _ = wait(futures, return_when=FIRST_COMPLETED)
+        else:
+            assert self._aip_execution_enabled
+            # TODO: This works for PoC, but maybe there is a better way
```

Comment:
Yeah, longer-term, maybe a cleaner way is to put each job in its own thread and wrap that thread in a regular concurrent future, so we can keep using `wait` here (and hopefully mix both parallel and AIP futures).